src/components/PostDetails.tsx

```diff
@@ -1,106 +1,154 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import * as serverAction from '../api/api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isCreateComment, setIsCreateComment] = useState(false);
+  const [isDeleteError, setIsDeleteError] = useState(false);
+
+  const loadComments = () => {
+    if (post && !isError) {
+      setIsCreateComment(false);
+      setIsLoading(true);
+
+      serverAction
+        .getComments(post.id)
+        .then(result => {
+          const results = result.map(comment => ({
+            id: comment.id,
+            postId: comment.postId,
+            name: comment.name,
+            body: comment.body,
+            email: comment.email,
+          }));
+
+          setComments(results);
+        })
+        .catch(() => {
+          setIsError(true);
+          setTimeout(() => {
+            setIsError(false);
+          }, 3000);
+        })
+        .finally(() => setIsLoading(false));
+    }
+  };
+
+  useEffect(() => {
+    loadComments();
+  }, [post, isError]);
```

Comment:
Including `isError` in the dependency array of `useEffect` might cause unnecessary re-fetching of comments when the error state changes. Consider removing `isError` from the dependencies to prevent this.